packages/core/bootstrap/src/index.ts

```diff
@@ -139,10 +139,11 @@ const executeSync = (execute: Execute, makeWsHandler?: MakeWSHandler): ExecuteSy
     withLogger,
     skipOnError(withCache),
     cacheWarmer.withCacheWarmer(storeSlice('cacheWarmer'), warmerMiddleware, { store: storeSlice('ws'), makeWSHandler: makeWsHandler })(execute),
-    ws.withWebSockets(storeSlice('ws'), makeWsHandler),
     rateLimit.withRateLimit(storeSlice('rateLimit')),
     withStatusCode,
-  ].concat(metrics.METRICS_ENABLED ? [withMetrics, withDebug] : [withDebug])
+  ]
+    .concat(metrics.METRICS_ENABLED ? [withMetrics, withDebug] : [withDebug])
+    .concat(ws.WS_ENABLED ? [ws.withWebSockets(storeSlice('ws'), makeWsHandler)] : [])
```

Comment:
You can check it inside the ws middleware better, so it would be like:
```js
if (!makeWsHandler || !WS_ENABLED) return await execute(input) // Middleware is ignored
// Middleware continues
```

Reply:
The order of these middlewares matters, so my suggestion is to move it back in its place and make the check inside the middleware. Also agree that the enable flag should be set internally in WS config.